### DISCUSSION TITLE
Why MobileWeb testing on a real mobile device connected to a Server is running fine, but taking 5X more time than mobile Browser emulation ?

### DISCUSSION THREAD

**Discussion by vnuta9** [appium/appium#20355] [Q&A]
I developed a test automation script which ran successfully on a web mobile emulator(desktop browser running with a pre-defined mobile device emulation. Only Chromium browsers supporting this feature as of now).

When I installed the code on a Desktop Server present in a Server room, while a 'Real Android mobile' device is connected via  a data cable and run the code, it is running fine in the real device as well. Only problem is it is running at a very slow pace, almost snail speed, taking more like the 5X amount of time, when compared to the previous browser-emulator. I did not have a nice work laptop, if in case, if I run it locally like real mobile devices are connected to it with data cable, it would have the same problem, I am guessing. Any ways, I would try this option in few more days.

BTW, the Real devices are accessing the company's eCommerce web site on it's mobile chrome browser(**mobileWeb testing**) using Wifi with >50mbps speed.

**My issue to resolve is:** `Why the test automation is running taking 5X more time on the Real device compared to the Browser based emulator ? I am using a decent data cable, not just a charging cable`

**Comment by KazuCocoa** [comment]
Debug level of chromedriver log would help to compare each step. For example the chromedriver itself gets different time duration by Chrome browser running on a desktop vs running on a mobile device.
A mobile emulator by a desktop could emulate network conditions, but I assume it does not emulate some resources like CPU. Some code in chromedriver has diff between for mobile and desktop thus chrome browser itself may have some differences between them.

**Comment by vnuta9** [comment]
@KazuCocoa 
Awesome fast reply. Most indebted to you !
Sorry, for another question!
**How do I set and obtain the different debug level logs on a ChromeDriver ?** (which I believe is downloaded by appium itself because, I have to run the appium with this command
`appium --allow-insecure chromedriver_autodownload`

BTW, I heard that mobile web browser is bit slower than the desktop browser, due to the nature of the platform on which each of them are running. But the difference in the time taken is **drastically higher**.

**Reply by KazuCocoa** [reply]
I assume these caps would help: `appium:showChromedriverLog` `appium:chromeLoggingPrefs` `appium:chromeOptions`
